### PR TITLE
fix build err while reading from stdin

### PR DESCRIPTION
**Pull request by zhangguanzhang**
fix https://github.com/containers/podman/issues/7107
Signed-off-by: zhangguanzhang <zhangguanzhang@qq.com>

### CODE REVIEW

**Comment by zhangguanzhang** [comment]
@rhatdan @TomSweeneyRedHat  
This unit test reading standard input to build Docker image issue,  which method should I use to write for it

File: cmd/podman/images/build.go [hunk 1]
```diff
@@ -229,7 +229,11 @@ func extractContextAndFiles(args, files []string) (string, []string, error) {
 		if utils.FileExists(filepath.Join(contextDir, "Containerfile")) {
 			containerFiles = append(containerFiles, filepath.Join(contextDir, "Containerfile"))
 		} else {
-			containerFiles = append(containerFiles, filepath.Join(contextDir, "Dockerfile"))
+			tmpDir := contextDir
```
Comment (TomSweeneyRedHat) [inline cmd/podman/images/build.go:232]: We want to leave in the Dockerfile appending as it was.  We want the build to default to using a Containerfile if present, if not, then to using a Dockerfile if present.   Perhaps add this above the else:

```
else if utils.FileExists(filepath.Join(contextDir, "Dockerrfile")) {
    containerFiles = append(containerFiles, filepath.Join(contextDir, "Dockerfile"))
} else {
```

**Comment by TomSweeneyRedHat** [comment]
@zhangguanzhang Thanks for taking this on, but please assign the issue to yourself if you're going to work on it.  I'll do so now.

**Comment by TomSweeneyRedHat** [comment]
Well it looks like you

**Comment by TomSweeneyRedHat** [comment]
Ugh, too many buttons, I was going to say @zhangguanzhang that you could not assign this to yourself after all.

**Comment by zhangguanzhang** [comment]
I have found the cause of the problem, and I will push to fix it later

**Comment by zhangguanzhang** [comment]
It work ok
```shell
$ /zgz# echo -e "FROM debian:testing\nRUN /bin/echo 'Test'" | ./podman/bin/podman build -t debian-test -  
STEP 1: FROM debian:testing
STEP 2: RUN /bin/echo 'Test'
Test
STEP 3: COMMIT debian-test
--> b9a6fcc86fa
b9a6fcc86fa6539d71ce82e63a92bf905726cafe2d37ca45939fff46e851fd37
```
but this unit test reading standard input to build Docker image issue, which method should I use to write for it

**Comment by TomSweeneyRedHat** [comment]
@zhangguanzhang I'm not sure which test you're having issues with?  Did you see @rhatdan 's #7121

**Comment by zhangguanzhang** [comment]
@rhatdan seems to have a more intuitive solution than mine, if his code merges please close this pr, or I will close this pr

**Comment by TomSweeneyRedHat** [comment]
@zhangguanzhang although this didn't get merged, I very much appreciate the quick response to the problem and your efforts to solve the problem!

**Pull request closed without merging**